src/utils/auth.js

```diff
@@ -0,0 +1,43 @@
+import { baseUrl, processRes } from "../utils/constants";
+
+export const signUp = (email, password, name) => {
+  return fetch(`${baseUrl}/signup`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email, password, name }),
+  })
+    .then((res) => processRes(res))
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(processRes)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.